plugins/woocommerce/client/blocks/assets/js/base/stores/woocommerce/cart.ts

```diff
@@ -142,6 +143,119 @@ const { state, actions } = store< Store >(
 					actions.showNoticeError( error as Error );
 				}
 			},
+			*batchAddCartItems( items: OptimisticCartItem[] ) {
+				const previousCart = JSON.stringify( state.cart );
+				const quantityChanges: QuantityChanges = {};
+
+				// Updates the database.
+				try {
+					const requests = items.map( ( item ) => {
+						const existingItem = state.cart.items.find(
+							( { id: productId } ) => item.id === productId
+						);
+
+						// Updates existing cart item.
+						if ( existingItem ) {
+							// Optimistically updates the number of items in the cart.
+							existingItem.quantity = item.quantity;
+							if ( existingItem.key ) {
+								quantityChanges.cartItemsPendingQuantity = [
+									existingItem.key,
+								];
+							}
+
+							return {
+								method: 'POST',
+								path: `/wc/store/v1/cart/update-item`,
+								headers: {
+									Nonce: state.nonce,
+									'Content-Type': 'application/json',
+								},
+								body: existingItem,
+							};
+						}
+
+						// Adds new cart item.
+						item = {
+							id: item.id,
+							quantity: item.quantity,
+							variation: item.variation,
+						} as OptimisticCartItem;
+						state.cart.items.push( item );
+						quantityChanges.productsPendingAdd =
+							quantityChanges.productsPendingAdd
+								? [
+										...quantityChanges.productsPendingAdd,
+										item.id,
+								  ]
+								: [ item.id ];
+
+						return {
+							method: 'POST',
+							path: `/wc/store/v1/cart/add-item`,
+							headers: {
+								Nonce: state.nonce,
+								'Content-Type': 'application/json',
+							},
+							body: item,
+						};
+					} );
+
+					const res: Response = yield fetch(
+						`${ state.restUrl }wc/store/v1/batch`,
+						{
+							method: 'POST',
+							headers: {
+								Nonce: state.nonce,
+								'Content-Type': 'application/json',
+							},
+							body: JSON.stringify( { requests } ),
+						}
+					);
+
+					const json: Cart = yield res.json();
```

Comment:
The type of this data is not `Cart`, but the following (see https://github.com/WordPress/wordpress-develop/blob/1651947e3dc6ba14e227cf5f2045df6c16908045/src/wp-includes/rest-api/class-wp-rest-server.php#L1816-L1822 and https://github.com/WordPress/wordpress-develop/blob/1651947e3dc6ba14e227cf5f2045df6c16908045/src/wp-includes/rest-api/class-wp-rest-server.php#L1862)
```ts
{
    responses: unknown[];
    failed: 'validation'
}
```